File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/actions/ArmAction.java
package org.firstinspires.ftc.teamcode.actions;

import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
import com.acmerobotics.roadrunner.Action;
import org.firstinspires.ftc.teamcode.robot.Arm;

public class ArmAction implements Action {
    private final Arm arm;
    private final Arm.Position armPosition;

    public ArmAction(Arm arm, Arm.Position armPosition) {
        this.arm = arm;
        this.armPosition = armPosition;
    }

    @Override
    public boolean run(TelemetryPacket p) {
        arm.setPosition(armPosition);
        return false;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Actions/ArmAction.kt
package org.firstinspires.ftc.teamcode.Actions

import com.acmerobotics.dashboard.telemetry.TelemetryPacket
import com.acmerobotics.roadrunner.Action
import org.firstinspires.ftc.teamcode.Robot.Arm
import org.firstinspires.ftc.teamcode.Robot.Wrist

class ArmAction(
    private val arm: Arm,
    private val armPosition: Arm.Position,
) : Action {

    override fun run(p: TelemetryPacket): Boolean {
        arm.setPosition(armPosition)
        return false
    }

}